docs/limitations.md

```diff
@@ -26,8 +26,17 @@ The [conditional execution](https://www.commonwl.org/v1.2/Workflow.html#Conditio
 
 ## Dealing with type incompatibility, e.g. mismatch of EDAM format
 
+Data types for tool inputs, and outputs, can (and should) be specified using the `format` field, as described in the [File Formats](https://www.commonwl.org/user_guide/16-file-formats/) tutorial. Doing so helps to ensure data consistency throughout a workflow, by checking type compatibility when data produced by one tool is used by another tool, as well as guiding users in what input data to provide (and what output data to expect). 
+
+It is recommended, where possible, to use a format definition taken from a widely-used ontological resource, such as [EDAM](http://edamontology.org). Doing so will enable users to reference the formats that your workflow expects, so that they can ensure they match these.
+
+Note, however, that the cwltool currently compares the format string only, it does not check the file for conformance to that format definition. It is up to you to make sure that you use the correct definition for your file format. If a format definition for your files is not available in the existing ontological databases, then you can set your own format string - however it is recommended that you do this in your own namespace, so it is clear it is not part of any existing database. It is also possible to request the addition of your format to a database, e.g. by raising an [issue for EDAM on github](https://github.com/edamontology/edamontology/issues).
+
 ## Managing file names and file storage
 
+CWL does ensure separation of working space for each step within a workflow, however output files cannot be renamed as they are gathered from steps, and the requested outputs are returned to the base directory from which the workflow is run. This can cause issues when using `scatter` to parallise a workflow, as outputs can potentially overwrite each other. It also can result in a messy workflow directory, if a lot of outputs are being returned.
```

Comment:
>  however output files cannot be renamed as they are gathered from steps

Sure, but they can be autonamed as part of the underlying `CommandLineTool` itself

> as outputs can potentially overwrite each other

That would be non-compliant with the CWL standards. Please remove this statement.

>  It also can result in a messy workflow directory
 
There is no concept in CWL of a `workflow directory`

Reply:
Thanks for the review :)

My comments here are influenced by the issues I had with organising the outputs from a workflow built on top of the BioBB building blocks (this is the main workflow file: [md_list.cwl](https://github.com/douglowe/biobb_hpc_cwl_md_list/blob/main/md_list.cwl)).

Their tool descriptors do not autoname the output files returned from the tool, so I had to write an extra tool to gather these outputs and place them into a directory with a unique name for each instance of the workflow. Otherwise the returned files would overwrite each other. The files were all returned to me in the same directory that I ran the workflow from - which is what I mean by `workflow directory`. Is there a way for a CWL user to specify a different directory for each `scatter` instance? I couldn't see anything that would do this when I was developing this workflow, but perhaps I missed the option somewhere?

Reply:
From our conversation yesterday: the results of a scatter are always kept in a different directory in the CWL reference runner and derivatives (as is required by the CWL standards). There was a bug for a while that when writing the final outputs some files with the same file name were clobbered; this should be fixed, so please test again and let us know which runner still has that problem.

Either way, please change the text of this section. 
```suggestion
CWL does ensure separation of working space for each step within a workflow, however output files cannot be renamed as they are gathered from steps. It can result in a messy final workflow output directory for some systems, if a lot of outputs are being returned.
```

Reply:
That makes sense to me. I'll correct this text now, and will test cwl-runner & toil to see how they deal with this kind of situation.

Reply:
This has been done in [PR 30](https://github.com/bioexcel/cwl-best-practice-guide/pull/30).